bluemix/configuration/core_config/bx_config.go

```diff
@@ -297,10 +297,11 @@ func (c *bxConfig) IAMID() (guid string) {
 }
 
 func (c *bxConfig) IsLoggedIn() (loggedIn bool) {
-	c.read(func() {
-		loggedIn = c.data.IAMToken != ""
-	})
-	return
+	if token := c.IAMToken(); token != "" {
+		iamTokenInfo := NewIAMTokenInfo(token)
+		return iamTokenInfo.HasExpired()
```

Comment:
We need to update this to check `!HasExpired()`, otherwise `IsLoggedIn()` will return `false` even if the token expiration is still valid. `HasExpired` returns true if the value of the token expiry is before the current time.

Reply:
You're right it is the opposite of what I have here.